bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/type/ZwaveJSTypeGeneratorImpl.java

```diff
@@ -0,0 +1,737 @@
+                property = Property.COLOR;
+            } else if (match(List.of("kelvin", "mirek", "mired"), details)) {
+                property = Property.COLOR_TEMPERATURE;
+            } else if (match(List.of("duration"), details)) {
+                property = Property.DURATION;
+            } else if (match(List.of("energy"), details)) {
+                property = Property.ENERGY;
+            } else if (match(List.of("fan"), details)) {
+                property = Property.SPEED;
+            } else if (match(List.of("freq"), details)) {
+                property = Property.FREQUENCY;
+            } else if (match(List.of("gas"), details)) {
+                property = Property.GAS;
+            } else if (match(List.of("location"), details)) {
+                property = Property.GEO_LOCATION;
+            } else if (match(List.of("heat"), details)) {
+                property = Property.HEATING;
+            } else if (match(List.of("conditioning", "cooling", " ac "), details)) {
+                property = Property.AIRCONDITIONING;
+            } else if (match(List.of("humidity"), details) && !match(List.of("soil"), details)) {
+                property = Property.HUMIDITY;
+            } else if (match(List.of("light"), details) && match(List.of("level"), details)) {
+                property = Property.ILLUMINANCE;
+            } else if (match(List.of("illuminance", " lux "), details)) {
+                property = Property.ILLUMINANCE;
+            } else if (match(List.of("battery"), details) && match(List.of("low", "alarm"), details)) {
+                property = Property.LOW_BATTERY;
+            } else if (Point.ALARM == point && match(List.of("battery"), details)) {
+                property = Property.LOW_BATTERY;
+            } else if (match(List.of("battery"), details)) {
+                property = Property.ENERGY;
+            } else if (match(List.of("media", "player", "television", " tv ", "receiver"), details)) {
+                property = Property.INFO;
+            } else if (match(List.of("soil"), details) && match(List.of("humidity"), details)) {
+                property = Property.MOISTURE;
+            } else if (match(List.of("moisture"), details)) {
+                property = Property.MOISTURE;
+            } else if (match(List.of("motion", " pir "), details)) {
+                property = Property.MOTION;
+            } else if (match(List.of("noise"), details)) {
+                property = Property.NOISE;
+            } else if (match(List.of(" oil "), details)) {
+                property = Property.OIL;
+            } else if (match(List.of("open", "close", " shut "), details)
+                    && match(List.of("state", "status"), details)) {
+                property = Property.OPEN_LEVEL;
+            } else if (match(List.of("open", "close", " shut "), details)
+                    && match(List.of("level", "position"), details)) {
+                property = Property.OPEN_LEVEL;
```

Comment:
Could be merged?

Reply:
They can be merged. As this mechanism does fuzzy detection, future refinements are to be expected. Therefor it was seperated. Changed for now.